nan.h

```diff
@@ -1654,12 +1661,12 @@ class Callback {
       Execute(progress);
   }
 
-  void SendProgress_(const char *data, size_t size) {
-    char *new_data = new char[size];
+  void SendProgress_(T *data, size_t size) {
+    T *new_data = new T[size];
     memcpy(new_data, data, size);
```

Comment:
Why not just do T *it = new T(*data);

On June 30, 2016 6:31:03 PM GMT+03:00, Ben Noordhuis <notifications@github.com> wrote:
>> @@ -1654,12 +1661,12 @@ class Callback {
>>        Execute(progress);
>>    }
>>  
>> -  void SendProgress_(const char *data, size_t size) {
>> -    char *new_data = new char[size];
>> +  void SendProgress_(T *data, size_t size) {
>> +    T *new_data = new T[size];
>>      memcpy(new_data, data, size);
>
>The simple solution would be to use `std::copy()` from `<algorithm>`,
>i.e.:
>```c++
>{
>  T *it = new_data;
>  std::copy(data, data + size, it);
>}
>```
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/588/files/85aa1f0be5361cd79a2c85f75cb6296188223c87#r69155126